src/parser/SGTypes.ts

```diff
@@ -359,6 +360,7 @@ export class SGInterface extends SGTag {
 
 export class SGComponent extends SGTag {
     constructor(
+        public ast: ElementCtx,
```

Comment:
Don't leak the internal XML parser ast; what information are you missing and how can it be defined in a generically useful way? Is it maybe the full range, including the open slash?

Reply:
With the brs parser AST, I have location information for almost every token. I want that same information available with the XML AST. It seems wasteful to pipe all of the tokens from the underlying parser into this, when we already have the ctx available. 

Ideally I would have the following info:
 - range for opening `<`
 - range for opening tag tag name
 - range for full attribute (i.e. `|someAttr="someval"|`)
 - range for attribute value (i.e. `someAttr="|someval|")
 - range for closing ">" in the opening tag
 - range for opening `<` in closing tag
 - range for closing tag name
 - range for closing `>` in closing tag

What practical benefit is there to hiding the internal XML parser AST? Is this to guard against the possible future change to a different xml parser?

Reply:
I really think the internal AST should not be used. It could be replaced, and it's unreliable.

We could have a few more ranges for XML tags, like the open and close tags, including angle brakets, they could even be late-resolution based on the internal AST.

Reply:
What do we propose to replace it with? Is it going to be more efficient that what we have now?

Reply:
@georgejecook we aren't proposing anything new at the moment. We are just designing our custom SceneGraph ast so it is resilient to these changes in the future. I agree with the idea. Once bsc lands at v1.0.0, we will be locked in to certain design decisions for a while, so I do want to have the conversations sooner rather than later about these types of things. 

@elsassph I like the idea of keeping the internal AST around, and exposing gettters. For on-demand access to the ranges I want. That's reasonable: it's efficient because we don't do unnecessary copying during parse, and it still gives us access to all the data from the original parser that's we might nee for any future getters that we don't expose right away. 

I'll go this route. Possibly in a separate PR since it's directly related to this PR.

Reply:
I removed this codeAction for now so the PR can merge. I'm still working on this in another branch and will submit a future PR when it's ready.